### PR TITLE
document certificate APIs

### PR DESCRIPTION
**Pull request by msaqib52**
This PR adds apiary documentation of required APIs for certificates.

### CODE REVIEW

**Comment by wajeeha-khalid** [comment]
@bradenmacdonald @ziafazal Please review

File: apiary.apib [hunk 1]
```diff
@@ -1907,6 +1907,23 @@ Allows for a caller to mark a user's notification as read, passed in by msg_id.
...
+### User Course Certificate Detail [GET]
+
+Returns a JSON representation of user certificate details for a specified course
+
++ Response 200 (application/json)
+
+        {
+            "status": "generated",
+            "web_view_uri": "http://lms.mcka.local/certificates/86927f12cc234e088c3f4663b2e00dc9",
+            "pdf_download_uri": "https://s3-eu-west-1.amazonaws.com/certificates/86927f12cc234e088c3f4663b2e00dc9/certificate.pdf",
```
Comment (ziafazal) [inline apiary.apib]: For html cert we would not have pdf_download_url.

File: apiary.apib [hunk 2]
```diff
@@ -3067,6 +3084,37 @@ Returns navigation information
         }
 
 
+## Course Certificates Status [/courses/{course_id}/certificates/status]
+
+It returns status for course certificates
+
+### Course Certificates Status [GET]
+
+Returns a JSON representation of certificate details for a specified course, also return status if we can request certificate generation for the course, possible statuses can be 'available', 'notavailable', 'generated', 'generating'
```
Comment (ziafazal) [inline apiary.apib]: Certificate status is per course per student. So we can create an API to return certificate status of learner in a particular course but there is not certificate status per course.

File: apiary.apib [hunk 3]
```diff
@@ -3067,6 +3084,37 @@ Returns navigation information
...
+
+        {
+            "status": "available",
+            "generation_uri": "http://lms.mcka.local/api/server/courses/course-v1:edX+DemoX+Demo_Course/certificates/generate",
+        }
+
+
+## Generate Course Certificates [/courses/{course_id}/certificates/generate]
+
+It starts generation of certificates fot the specified course
```
Comment (bradenmacdonald) [inline apiary.apib]: What's our use case here: Do we intend to call this once per course at some specific "end of course" date, or call this many times on an ongoing basis?

Comment (msaqib52) [inline apiary.apib]: we will call this once at the end of the course to generate certificates for all the participants for the specified course.

File: apiary.apib (hunk 2 above)
Comment (msaqib52) [inline apiary.apib]: Our use case is to display a button to apros admin if course certificates are available i-e course end date or any other criteria met if any and change the status of the button accordingly. Apros admin would click this button to generate certificates for all participants for the specified course.

File: apiary.apib (hunk 1 above)
Comment (msaqib52) [inline apiary.apib]: pdf_download_uri removed

**Comment by msaqib52** [comment]
closing this PR in favour of #44

**Pull request closed without merging**